decoder/stats.go

```diff
@@ -198,15 +205,26 @@ func updateEncounterStats(pokemon *Pokemon) {
 		pokemonStatsLock.Lock()
 		defer pokemonStatsLock.Unlock()
 
-		countStats := pokemonCount[areaName]
-		if countStats == nil {
-			countStats = &areaPokemonCountDetail{}
+		countStats, exists := pokemonCount[areaName]
+		if !exists {
+			countStats = &areaPokemonCountDetail{
+				hundos:      make(map[pokemonForm]int),
```

Comment:
Do you know how many elements are put in this map? you can pre-size it to save growths (that probably are inevitable)

Reply:
I don't have much data, since that's the first day I run it on prod. But tbh, is there a point of tuning this place that much... ? It was overengineered and ugly as hell to read.

```sql
SELECT COUNT(DISTINCT pokemon_id, form_id) FROM `pokemon_iv_stats` where date = '2025-03-08'
-- 347
```

Reply:
It's a standard thing to do to correctly size maps (and slices) since they expand and cause GC. It's a couple of characters to do so.

Reply:
Well yeah I can add, but would need to gather more data I guess?

Reply:
I was just googling for the boundaries and it appears go 1.24 has switched map implementation so this is likely not needed, so ignore this change